qiskit_nature/problems/sampling/protein_folding/peptide/chains/main_chain.py

```diff
@@ -0,0 +1,189 @@
+                                        beads.
+
+        Raises:
+            InvalidSizeException: when the length of list of side chain lengths provided does not
+                                    equal the length of the main chain.
+            InvalidSideChainException: when first, second or last main beads have a side chain.
+        """
+        self._main_chain_residue_sequence = main_chain_residue_sequences
+        beads_list = self._build_main_chain(
+            main_chain_len,
+            main_chain_residue_sequences,
+            side_chain_lens,
+            side_chain_residue_sequences,
+        )
+        super().__init__(beads_list)
+
+    @property
+    def main_chain_residue_sequence(self) -> List[str]:
+        """Returns a residue sequence for the main chain."""
+        return self._main_chain_residue_sequence
+
+    def _build_main_chain(
+        self,
+        main_chain_len: int,
+        main_chain_residue_sequences: List[str],
+        side_chain_lens: List[int],
+        side_chain_residue_sequences: List[str],
+    ) -> Sequence[MainBead]:
+        """
+        Creates a main chain for a given main chain length and side chain data.
+
+        Args:
+            main_chain_len: length of the main chain of a peptide.
+            main_chain_residue_sequences: list of characters that define residues for a main chain.
+            side_chain_lens: list of lengths of all side chains.
+            side_chain_residue_sequences: list of characters that define residues for all side
+            beads.
+
+        Returns:
+            An instance of a MainChain class.
+
+        Raises:
+            InvalidSizeException: when the length of list of side chain lengths provided does not
+                                    equal the length of the main chain.
+            InvalidSideChainException: when first, second or last main beads have a side chain.
+        """
+        main_chain = []
+        self._validate_chain_lengths(main_chain_len, side_chain_lens)
+        self._validate_side_chain_index_by_lengths(side_chain_lens)
+        self._validate_side_chain_index_by_residues(side_chain_residue_sequences)
+
+        for main_bead_id in range(main_chain_len - 1):
+            bead_turn_qubit_1 = self._build_turn_qubit(main_chain_len, 2 * main_bead_id)
+            bead_turn_qubit_2 = self._build_turn_qubit(main_chain_len, 2 * main_bead_id + 1)
+            side_chain = self._create_side_chain(
+                main_bead_id, main_chain_len, side_chain_lens, side_chain_residue_sequences
+            )
+            main_bead = MainBead(
+                main_bead_id,
+                main_chain_residue_sequences[main_bead_id],
+                [bead_turn_qubit_1, bead_turn_qubit_2],
+                side_chain,
+            )
+            main_chain.append(main_bead)
+        main_bead = MainBead(main_chain_len - 1, None, None, None)
```

Comment:
Is this simply an end-marker?

Reply:
Yes. What we mostly care about are qubits that encode turns and there are 1 less turns than beads. I added the dummy bead at the end that stores no turn to be consistent with the length of a peptide that is defined in the number of beads.